File: ILT.IHR.DTO/Appraisal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public partial class Appraisal : AbstractDataObject
    {
        public int AppraisalID
        {
            get { return base.RecordID; }
            set { base.RecordID = value; }
        }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int ReviewYear { get; set; }
        public int? EmployeeID { get; set; }
        public string EmployeeName { get; set; }
        public int? FinalReviewerID { get; set; }
        public string Manager { get; set; }
        public string FinalReviewer { get; set; }
        public int? ReviewerID { get; set; }
        public string Reviewer { get; set; }
        public DateTime? AssignedDate { get; set; }
        public DateTime? SubmitDate { get; set; }
        public DateTime? ReviewDate { get; set; }
        public DateTime? FinalReviewDate { get; set; }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int StatusID { get; set; }
        public string Status { get; set; }
        public string StatusValue { get; set; }
        public string MgrFeedback { get; set; }
        public string Comment { get; set; }
   
        public List<AppraisalDetail> AppraisalDetails { get; set; }
        public List<AppraisalGoal> AppraisalGoals { get; set; }
    }
}

File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/Appraisal.ts
import { IAppraisalDetail, IAppraisalDetailDisplay } from "./AppraisalDetail";
import { IAppraisalGoal, IAppraisalGoalDisplay } from "./AppraisalGoal";

export class IAppraisal {
    AppraisalID: number;
    ReviewYear: number;
    EmployeeID?: number;
    EmployeeName?: string;
    FinalReviewerID?: number;
    Manager?: string;
    FinalReviewer?: string;
    ReviewerID?: number;
    Reviewer?: string;
    AssignedDate?: Date;
    SubmitDate?: Date;
    ReviewDate?: Date;
    FinalReviewDate?: Date;
    StatusID: number;
    Status?: string;
    StatusValue?: string;
    MgrFeedback: string;
    Comment: string;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: [''];
    //AppraisalDetails: IAppraisalDetail[];
    //AppraisalGoal: IAppraisalGoal[];
}
export class IAppraisalDisplay {
    appraisalID: number;
    reviewYear: number;
    employeeID?: number;
    employeeName?: string;
    finalReviewerID?: number;
    manager?: string;
    finalReviewer?: string;
    reviewerID?: number;
    reviewer?: string;
    assignedDate?: Date;
    submitDate?: Date;
    reviewDate?: Date;
    finalReviewDate?: Date;
    statusID: number;
    status?: string;
    statusValue?: string;
    mgrFeedback: string;
    comment: string;
    createdBy: string;
    createdDate: Date;
    modifiedBy: string;
    modifiedDate: Date;
    rimeStamp: ['']
    appraisalDetails: IAppraisalDetailDisplay[]
    appraisalGoals: IAppraisalGoalDisplay[]
}
